platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py

```diff
@@ -85,194 +52,156 @@ def qsfp_presence_get(client):
+
+    def get_name(self):
+        """
+        Retrieves the name of the device
+            Returns:
+            string: The name of the device
+        """
+        return "sfp{}".format(self.index)
+
+    def get_reset_status(self):
+        """
+        Retrieves the reset status of SFP
+        """
+        def get_qsfp_reset(pltfm_mgr):
+            return pltfm_mgr.pltfm_mgr_qsfp_reset_get(self.index)
+        _, status = pltfm_mgr_try(get_qsfp_reset, False)
+        return status
 
     def reset(self):
-        return Sfp.sfputil.reset(self.port_num)
-
-    def get_transceiver_info(self):
-        with Sfp.sfputil.eeprom_action() as u:
-            return u.get_transceiver_info_dict(self.port_num)
-
-    def get_transceiver_bulk_status(self):
-        with Sfp.sfputil.eeprom_action() as u:
-            return u.get_transceiver_dom_info_dict(self.port_num)
+        """
+        Reset SFP and return all user module settings to their default srate.
+        """
+        def qsfp_reset(client):
+            client.pltfm_mgr.pltfm_mgr_qsfp_reset(self.index, True)
+            return client.pltfm_mgr.pltfm_mgr_qsfp_reset(self.index, False)
 
-    def get_transceiver_threshold_info(self):
-        with Sfp.sfputil.eeprom_action() as u:
-            return u.get_transceiver_dom_threshold_info_dict(self.port_num)
+        err = thrift_try(qsfp_reset)
+        return not err
 
-    def get_change_event(self, timeout=0):
-        return Sfp.get_transceiver_change_event(timeout)
+    def get_status(self):
+        """
+        Retrieves the operational status of the device
+        """
+        reset = self.get_reset_status()
 
-def sfp_list_get():
-    sfp_list = []
-    for index in range(Sfp.port_start(), Sfp.port_end() + 1):
-        sfp_node = Sfp(index)
-        sfp_list.append(sfp_node)
-    return sfp_list
+        if reset:
+            status = False
+        else:
+            status = True
+
+        return status
+
+    def get_position_in_parent(self):
+        """
+        Retrieves 1-based relative physical position in parent device.
+        Returns:
+            integer: The 1-based relative physical position in parent
+            device or -1 if cannot determine the position
+        """
+        return self.index
+
+    def is_replaceable(self):
+        """
+        Indicate whether this device is replaceable.
+        Returns:
+            bool: True if it is replaceable.
+        """
+        return True
+
+    def get_error_description(self):
+        """
+        Retrives the error descriptions of the SFP module
+        Returns:
+            String that represents the current error descriptions of vendor specific errors
+            In case there are multiple errors, they should be joined by '|',
+            like: "Bad EEPROM|Unsupported cable"
+        """
+        if not self.get_presence():
+            return self.SFP_STATUS_UNPLUGGED
+        return self.SFP_STATUS_OK
+
+    def tx_disable(self, tx_disable):
```

Comment:
tx_disable is not platform specific. Why override? its already available here [https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/sff8436.py#L227](https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/sff8436.py#L227)